cli/azd/ci-build.ps1

```diff
@@ -39,43 +48,94 @@ if ($IsWindows) {
+$buildFlags = @(
+    # Remove file system paths from the compiled binary
+    "-gcflags=-trimpath",
+    # Remove file system paths from the assembled code
+    "-asmflags=-trimpath",
+    # remove all file system paths from the resulting executable.
+    # Instead of absolute file system paths, the recorded file names
+    # will begin either a module path@version (when using modules),
+    # or a plain import path (when using the standard library, or GOPATH).
+    "-trimpath"
+)
+
+if ($CodeCoverageEnabled) {
+    $buildFlags += "-cover"
+}
+
+# Build constraint tags
+# cfi: Enable Control Flow Integrity (CFI),
+# cfg: Enable Control Flow Guard (CFG),
+# osusergo: Optimize for OS user accounts
+$tagsFlag = "-tags=cfi,cfg,osusergo"
+
+# ld linker flags
+# -s: Omit symbol table and debug information
+# -w: Omit DWARF symbol table
+# -X: Set variable at link time. Used to set the version in source.
+$ldFlag = "-ldflags=`"-s -w -X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)' "
+
 if ($IsWindows) {
-    Write-Host "go build (windows)"
-    go build `
-        -buildmode=exe `
-        -tags="cfi,cfg,osusergo" `
-        -trimpath `
-        -gcflags="-trimpath" `
-        -asmflags="-trimpath" `
-        -ldflags="-s -w -X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)' -linkmode=auto -extldflags=-Wl,--high-entropy-va"
+    Write-Host "Building for windows"
+    $buildFlags += @(
+        "-buildmode=exe",
+        $tagsFlag,
+        # -extldflags=-Wl,--high-entropy-va: Pass the high-entropy VA flag to the linker to enable high entropy virtual addresses
+        ($ldFlag + "-linkmode=auto -extldflags=-Wl,--high-entropy-va`"")
+    )
 }
 elseif ($IsLinux) {
-    Write-Host "go build (linux)"
-    go build `
-        -buildmode=pie `
-        -tags="cfi,cfg,cfgo,osusergo" `
-        -gcflags="-trimpath" `
-        -asmflags="-trimpath" `
-        -ldflags="-s -w -X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)' -extldflags=-Wl,--high-entropy-va"
+    Write-Host "Building for linux"
+    $buildFlags += @(
+        "-buildmode=pie",
+        ($tagsFlag + ",cfgo"),
+        # -extldflags=-Wl,--high-entropy-va: Pass the high-entropy VA flag to the linker to enable high entropy virtual addresses
+        ($ldFlag + "-extldflags=-Wl,--high-entropy-va`"")
+    )
 }
 elseif ($IsMacOS) {
-    Write-Host "go build (macos)"
-    go build `
-        -buildmode=pie `
-        -tags="cfi,cfg,cfgo,osusergo" `
-        -gcflags="-trimpath" `
-        -asmflags="-trimpath" `
-        -ldflags="-s -w -X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)' -linkmode=auto"
+    Write-Host "Building for macOS"
+    $buildFlags += @(
+        "-buildmode=pie",
+        ($tagsFlag + ",cfgo"),
+        # -linkmode=auto: Link Go object files and C object files together
+        ($ldFlag + "-linkmode=auto`"")
+    )
 }
```

Comment:
I found it much easier to read & understand the OS specific build command/flags in use prior to this change.  This update requires additional cognitive load to decipher what should ideally be a simple build command.

Reply:
Moving it to #2005 that is meant to have the changes checked in.